### DISCUSSION TITLE
When using the translator `cs` defaults to `en` instead of `cs_CZ`

### DISCUSSION THREAD

**Discussion by SavageTiger** [symfony/symfony#51757] [Q&A]
He everyone 👋 

I have a question about the translator. In our application we usually only ask the language instead of all the dialects.

When I do this `$translator->trans('translation.key', [], 'messages', 'nl')` the translator understands that it should use `nl_NL` and it will not default back to `en`.

The only configuration that we have on our translator is that it defaults to `en`

This usually work except for `czech` (in our case). Because the locale for `czech` isn't `cz_CZ` but `cs_CZ` when I do this `$translator->trans('translation.key', [], 'messages', 'cs')` it does seem to default to `en`.

This feels like a bug, because it should understand (if you ask me) that `cs_CZ` is closer to `cs` then `en_GB`.

Maybe I am really misguided, but it would be nice to hear others perspective.

Thank in advance!

**Comment by stof** [comment]
Translating `cs` will never fallback to `cs_CZ`. The fallback is done the other way: when asked to translate something into `cs_CZ` (the country-specific variant of Czech), it can fallback to `cs` (the generic Czech locale).

So if you want to configure locales without country variants when calling the translator, you should _define_ translations for the generic locales.

**Reply by ruudk** [reply]
The fallback works like that indeed. But does that make sense from a user perspective?

Why would `en` be preferred over the country specific `cs_CZ` when the `cs` was given.

**Reply by MatTheCat** [reply]
Because

- you did not configure any of your translations as `cs`
- you configured the default locale as `en`

So it really is the developer’s responsibility.
